lib/router/src/utils.ts

```diff
@@ -37,27 +38,48 @@ interface Args {
   [key: string]: any;
 }
 
-const deepDiff = (value: any, update: any): any => {
-  if (deepEqual(value, update)) return undefined;
+export const DEEPLY_EQUAL = Symbol('Deeply equal');
+export const deepDiff = (value: unknown, update: unknown): any => {
   if (typeof value !== typeof update) return update;
-  if (Array.isArray(value)) return update;
-  if (update && typeof update === 'object') {
-    return Object.keys(update).reduce((acc, key) => {
-      const diff = deepDiff(value[key], update[key]);
-      return diff === undefined ? acc : Object.assign(acc, { [key]: diff });
+  if (deepEqual(value, update)) return DEEPLY_EQUAL;
+  if (Array.isArray(value) && Array.isArray(update)) {
+    return value.length > update.length
```

Comment:
I wonder if an additional "early return if" might be more readable than a multi-line ternary:
`if (update.length >= value.length) return update;`

Reply:
Good call.